backend/src/services/InterviewService.ts

```diff
@@ -0,0 +1,139 @@
+import { type Server as HTTPServer } from "http";
+
+import { Server } from "socket.io";
+
+import env from "../env";
+import { InterviewModel, type InterviewState } from "../models/InterviewModel";
+import { ReviewModel } from "../models/ReviewModel";
+
+interface Payload {
+  userId: string;
+  key: string;
+  value: string | boolean | number;
+}
+
+interface SelectionPayload {
+  role: number;
+  from: number;
+  to: number;
+}
+
+const questionPlaceholder = `# TSE Technical Interview
```

Comment:
Can we fetch this dynamically from [here](https://raw.githubusercontent.com/TritonSE/TSE-Technical-Interview-Template/main/README.md)? Maybe we can fetch again every time (in other words, don't cache it), so that we can update the template on GitHub and not need to restart the backend before future interviews.

I think we're technically not supposed to use GitHub as a CDN, but the load should be low enough to not matter :laughing:

Reply:
Good idea, implemented -- I changed it to fetch the readme at startup, then fetch it again for each new interview (so if we get blocked at some point the running instance will still have something as backup).  The Promise I wrote is a bit weird because of it (resolves with `null` instead of rejecting), but it works well.